graphql_compiler/schema_transformation/utils.py

```diff
@@ -39,16 +40,54 @@ class InvalidTypeNameError(SchemaTransformError):
     """
 
 
-class SchemaNameConflictError(SchemaTransformError):
-    """Raised when renaming or merging types or fields cause name conflicts.
+class SchemaMergeNameConflictError(SchemaTransformError):
+    """Raised when merging types or fields cause name conflicts.
 
-    This may be raised if a field or type is renamed to conflict with another field or type,
-    if two merged schemas share an identically named field or type, or if a
+    This may be raised if two merged schemas share an identically named field or type, or if a
     CrossSchemaEdgeDescriptor provided when merging schemas has an edge name that causes a
     name conflict with an existing field.
     """
 
 
+class SchemaRenameNameConflictError(SchemaTransformError):
+    """Raised when renaming types or fields cause name conflicts.
+
+    This may be raised if a field or type is renamed to conflict with another field or type.
+    """
+
+    name_conflicts: Dict[str, Set[str]]
+
+    def __init__(
+        self, name_conflicts: Dict[str, Set[str]], renamed_to_scalar_conflicts: Dict[str, str]
+    ) -> None:
+        """Record all renaming conflicts."""
+        super().__init__()
+        self.name_conflicts = name_conflicts
+        self.renamed_to_scalar_conflicts = renamed_to_scalar_conflicts
+
+    def __str__(self) -> str:
+        """Explain renaming conflict and the fix."""
+        name_conflicts_message = ""
+        if self.name_conflicts != {}:
+            name_conflicts_message = (
+                f"Applying the renaming would produce a schema with multiple types with the same "
```

Comment:
I find the wording for these two messages a bit awkward, in particular these parts:

> Each entry here shows the type name that would appear in the renamed schema, along with all type names in the original schema that map to it

> it does not map a type name to a name that a different type will have in the resulting schema

> name already taken by a scalar

Reply:
For the first,
`Received the following name conflicts: {self.name_conflicts}, where each entry maps a type in the renamed schema ("key") to the set of types in the original schema that would be renamed to "key".`
^ oof it is really tough to write this clearly, but this is my best attempt...

For the second, I think you can eliminate that part entirely because it is just rephrasing `ensure that no two types in the renamed schema have the same name` in a more confusing way. Instead,
`To fix this, modify the renamings argument of rename_schema to ensure that no two types in the renamed schema have the same name.`

I don't actually think the third sounds too bad, but maybe `used by` would sound better than `taken by`?

Reply:
Updated, how does this sound?